### ISSUE TITLE
Add some logic for defaultFSharpBinariesDir when initialization of FsiEvaluationSession

### ISSUE THREAD

**Issue by ingted** [dotnet/fsharp#11293]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated to add customized code for the null result from FSharpEnvironment.tryCurrentDomain()
(to integrate FSharp API of Akka.net with pythonnet)

**Describe the solution you'd like**
E.g. add something like this...

```
        #if INTERACTIVE
            let defaultFSharpBinariesDir = AppDomain.CurrentDomain.BaseDirectory
        #else
            let defaultFSharpBinariesDir =
                let d = System.Reflection.Assembly.GetExecutingAssembly()     
                let fi = IO.FileInfo(d.Location)
                fi.DirectoryName  
        #endif
        defaultFSharpBinariesDir 
```

**Comment by cartermp** [comment]
I'm not confident we'd do this ourselves, but we'd be happy to take a look at a PR.

**Comment by ingted** [comment]
https://github.com/dotnet/fsharp/pull/11309

**Comment by vzarytovskii** [comment]
This should be fixed by #11824
